target/linux/ramips/files/drivers/net/ethernet/ralink/esw_rt3050.c

```diff
@@ -1441,12 +1441,8 @@ static int esw_probe(struct platform_device *pdev)
 	if (reg_init)
 		esw->reg_led_source = be32_to_cpu(*reg_init);
 
-	esw->rst_esw = devm_reset_control_get(&pdev->dev, "esw");
-	if (IS_ERR(esw->rst_esw))
-		esw->rst_esw = NULL;
-	esw->rst_ephy = devm_reset_control_get(&pdev->dev, "ephy");
-	if (IS_ERR(esw->rst_ephy))
-		esw->rst_ephy = NULL;
+	esw->rst_esw = devm_reset_control_get_optional_exclusive(&pdev->dev, "esw");
```

Comment:
I think `esw` reset can be removed.

Reply:
I need to double-check how that's handled internally. Will test in the evening.

Reply:
> I think `esw` reset can be removed.

Are you sure that this is true? I'm studying documentation for reset framework and device tree bindings, and it looks to me that current implementation is correct, as the resets are still matched by their names here - but the sequencing handling by reset framework differs if we specify those reset this way. IMHO this should stay like that.

Reply:
Let me explain why I think we need to directly delete these codes. The vendor driver said, `RT5350/MT7628 (SDMA) need to reset ESW and FE at the same to avoid PDMA panic`. My understanding is that none of them (esw && fe) can be individually reset. So we don't need it at all here. I believe that any device only needs to be reset once after booting up. That's also the vendor driver behavior. I've tested it's safe to remove it on MT7628.

BTW, only RT305x/RT335x/RT5350/MT7628 use this driver.

Reply:
Fixed now.